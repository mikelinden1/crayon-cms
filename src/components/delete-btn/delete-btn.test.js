import DeleteBtn from './delete-btn';
import sinon from 'sinon';

const props = {
    item: {
        name: 'test'
    },
    config: {
        capabilities: {
            deleteable: true
        }
    },
     actions: {
         deleteItem: () => {}
     }
};

it('DeleteBtn matches snapshot', () => {
    const wrapper = shallow(<DeleteBtn {...props} />);
    expect(wrapper).toMatchSnapshot();
});

it('DeleteBtn doesn\'t display if we don\'t have an archive config', () => {
    const propsNoDelete = {
        ...props,
        config: {
            capabilities: {
                deleteableu: false
            }
        }
    };

    const wrapper = shallow(<DeleteBtn {...propsNoDelete} />);
    expect(wrapper.type()).toEqual(null)
});

it('DeleteBtn calls deleteItem and sends the item', () => {
    const spy = sinon.spy();
    const propsWithSpy = {
        ...props,
        actions: {
            deleteItem: spy
        }
    };

    const wrapper = mount(<DeleteBtn {...propsWithSpy} />);
    wrapper.find('Button').first().simulate('click');

    expect(spy.calledOnce).toBeTruthy();
    expect(spy.getCall(0).args[0]).toEqual(propsWithSpy.item);
});

it('DeleteBtn shows a spinner when deleting', () => {
    const spy = sinon.spy();
    const propsWithDeleting = {
        ...props,
        item: {
            ...props.item,
            deleting: true
        }
    };

    const wrapper = shallow(<DeleteBtn {...propsWithDeleting} />);
    expect(wrapper).toMatchSnapshot();
});