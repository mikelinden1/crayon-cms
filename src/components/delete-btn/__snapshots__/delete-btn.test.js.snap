// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteBtn matches snapshot 1`] = `
<Button
  color="danger"
  onClick={[Function]}
  tag="button"
>
  Delete
</Button>
`;

exports[`DeleteBtn shows a spinner when deleting 1`] = `
<Button
  color="danger"
  disabled={true}
  tag="button"
>
  <Spinner />
</Button>
`;
